pkg/virt-handler/vm.go

```diff
@@ -151,6 +153,13 @@ type VirtualMachineController struct {
 	podIsolationDetector     isolation.PodIsolationDetector
 	containerDiskMounter     *container_disk.Mounter
 	clusterConfig            *virtconfig.ClusterConfig
+
+	// records if pod network initial-config has completed
+	// initial-config involves cycling an entire posix thread
+	// so for performance, knowing phase1 is complete
+	// prevents cycling an unncessary posix thread.
+	InitialNetworkConfigCache     map[types.UID]int
```

Comment:
I'm not actually sure that it involves cycling the posix thread. I want to be sure that adding the lock is necessary. How can I do that?